### DISCUSSION TITLE
How to use a theme that's not available in this package, but available in Highlight.js?

### DISCUSSION THREAD

**Discussion by sebpowell** [react-syntax-highlighter/react-syntax-highlighter#523] [Q&A] [answered]
Hello, hoping someone might be able to give me some direction. 

I would like to use the Highlight.js theme `Tokyo Night Dark`. As far as I can see, it isn't included in this package, so I'm wondering what the best way of adding it in is. 

Highlight.js provide its themes in a CSS format, whereas `react-syntax-highlightter` provides them in a JS format. So is the right way to go about this to convert the Highlight.js theme into the same format that `react-syntax-highlighter` uses and provide it to the component that way?

I ask because the alternative would be to wrap the component in one which has the Highlight.js CSS attached to it (which would save having to convert the CSS to a JS format). However, as far as I can tell, the style property for the component will fallback to a default theme if a prop isn't provided, meaning this approach won't, in theory, work. 

Has anyone else come across this issue? Any pointers would be greatly appreciated.

**Comment by schrodienieur** [comment, marked as answer]
I think the best way to add theme is by converting the css into js, then make a pull request. 

I have tried to convert the tokyo-night-dark.css to js format from [https://github.com/highlightjs/highlight.js/blob/main/src/styles/tokyo-night-dark.css](https://github.com/highlightjs/highlight.js/blob/main/src/styles/tokyo-night-dark.css)

And here is the result:

`tokyo-night-dark.js`
```
export default {
    "hljs": {
        "display": "block",
        "overflowX": "auto",
        "padding": "0.5em",
        "color": "#9aa5ce",
        "background": "#1a1b26"
    },
    "hljs-meta": {
        "color": "#565f89"
      },
    "hljs-comment": {
        "color": "#565f89"
    },
    "hljs-tag": {
        "color": "#f7768e"
    },
    "hljs-doctag": {
        "color": "#f7768e"
    },
    "hljs-selector-id": {
        "color": "#f7768e"
    },
    "hljs-selector-class": {
        "color": "#f7768e"
    },
    "hljs-regexp": {
        "color": "#f7768e"
    },
    "hljs-template-tag": {
        "color": "#f7768e"
    },
    "hljs-selector-pseudo": {
        "color": "#f7768e"
    },
    "hljs-selector-attr": {
        "color": "#f7768e"
    },
    "hljs-variable.language_": {
        "color": "#f7768e"
    },
    "hljs-deletion": {
        "color": "#f7768e"
    },
    "hljs-variable": {
        "color": "#ff9e64"
    },
    "hljs-template-variable": {
        "color": "#ff9e64"
    },
    "hljs-number": {
        "color": "#ff9e64"
    },
    "hljs-literal": {
        "color": "#ff9e64"
    },
    "hljs-type": {
        "color": "#ff9e64"
    },
    "hljs-params": {
        "color": "#ff9e64"
    },
    "hljs-link": {
        "color": "#ff9e64"
    },
    "hljs-built_in": {
        "color": "#e0af68"
    },
    "hljs-attribute": {
        "color": "#e0af68"
    },
    "hljs-keyword": {
        "color": "#7dcfff",
    },
    "hljs-title.function_": {
        "color": "#7dcfff"
    },
    "hljs-title": {
        "color": "#7dcfff",
    },
    "hljs-title.class_": {
        "color": "#7dcfff"
    },
    "hljs-title.class_.inherited__": {
        "color": "#7dcfff"
    },
    "hljs-subst": {
        "color": "#7dcfff"
    },
    "hljs-property": {
        "color": "#7dcfff"
    },
    "hljs-selector-tag": {
        "color": "#73daca"
    },
    "hljs-quote": {
        "color": "#9ece6a"
    },
    "hljs-string": {
        "color": "#9ece6a"
    },
    "hljs-symbol": {
        "color": "#9ece6a"
    },
    "hljs-bullet": {
        "color": "#9ece6a"
    },
    "hljs-addition": {
        "color": "#9ece6a"
    },
    "hljs-code": {
        "color": "#7aa2f7"
    },
    "hljs-formula": {
        "color": "#7aa2f7"
    },
    "hljs-section": {
        "color": "#7aa2f7"
    },
    "hljs-name": {
        "color": "#bb9af7"
    },
    "hljs-keyword": {
        "color": "#bb9af7"
    },
    "hljs-operator": {
        "color": "#bb9af7"
    },
    "hljs-char_escape_": {
        "color": "#bb9af7"
    },
    "hljs-attr": {
        "color": "#bb9af7"
    },
    "hljs-punctuation": {
        "color": "#c0caf5"
    },    
    "hljs-emphasis": {
        "fontStyle": "italic"
    },
    "hljs-strong": {
        "fontWeight": "bold"
    }
}
```

Save and import it

```
import tokyoNightDark from 'react-syntax-highlighter/dist/esm/styles/hljs/tokyo-night-dark';
```

But, there are some css that i don't know how to convert it, like below

 There are multiple value of .hljs-keyword (#bb9af7 and #7dcfff)
 I end up with writing all the values.
 
 Then, for .hljs-title.function_, .hljs-variable.language_, .hljs-title.class_, .hljs-title.class_.inherited__  , i dont know how to convert it.
  
  I write like this below (got it from ChatGPT), but seems does not work
```
"hljs-title.function_": {
    "color": "#7dcfff"
}
```

Compared to the [demo](https://highlightjs.org/static/demo/), if you look at Javascript demo, some code are not highlighted such as search and checkCondition function, that has "hljs-title function_" className.

**Reply by sebpowell** [reply]
@mavic111 thanks! I ended up doing something similar to this, also opened up a PR with the result for those who may want to use it: https://github.com/react-syntax-highlighter/react-syntax-highlighter/pull/525/files
